### PR TITLE
Switch from plain unittest to pytest

### PR DESCRIPTION
**Pull request by berquist**
Following https://github.com/cclib/cclib/pull/485, this is the remainder of https://github.com/cclib/cclib/pull/456. The output is not pretty right now, especially `test_data` which is not expanded, but everything is being tested.

### CODE REVIEW

**Comment by berquist** [comment]
I'm going to try and figure out how to have the output from `test_all` and the visual tests displayed. Then we should have basic parity with the current setup.

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import sys
+
+
+version_major = sys.version_info.major
+version_minor = sys.version_info.minor
+
+# Paths that should be ignored for all Python versions.
+paths_allver = [
```
Comment (langner) [inline conftest.py]: Maybe call this `paths_ignore_allver`?

File: conftest.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import sys
```
Comment (langner) [inline conftest.py]: Should this file go in the `test/` directory?

File: pytest.ini [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[pytest]
```
Comment (langner) [inline pytest.ini]: Does this file need to go into the top level directory?

File: test/conftest.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import sys
```
Comment (langner) [inline test/conftest.py]: Where is this module used?

Comment (berquist) [reply test/conftest.py]: The version check on line 4,5: https://github.com/cclib/cclib/pull/491/files#diff-f57b4d3a4403d70b647003618269deeaR4

Comment (langner) [reply test/conftest.py]: No, I mean where is `conftest` used :)

Comment (langner) [reply test/conftest.py]: I see. Could you add some comment to the file (perhaps in the docstring) explaining this?

File: test/regression.py [hunk 5]
```diff
@@ -1772,7 +1772,7 @@ def old_unit_test(logfile):
     return old_unit_test
 
 
-def main(which=[], opt_traceback=False, opt_status=False, regdir=__regression_dir__):
+def test_regressions(which=[], opt_traceback=False, opt_status=False, regdir=__regression_dir__):
```
Comment (langner) [inline test/regression.py]: Why do we need to rename this?

File: test/data/testGeoOpt.py [hunk 6]
```diff
@@ -44,7 +44,8 @@ def testatomnos(self):
         # This will work only for numpy
         #self.assertEquals(self.data.atomnos.dtype.char, 'i')
 
-        atomnos_types = [numpy.issubdtype(atomno,int) for atomno in self.data.atomnos]
+        atomnos_types = [numpy.issubdtype(atomno, np.signedinteger)
```
Comment (langner) [inline test/data/testGeoOpt.py]: This is fine, but for the future generally it's better to separate no-op formatting changes from actual changes.

File: test/conftest.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+]
+
+
+def match_path(path, partial_paths):
+    """Does the given path contain any of the stubs in partial_paths?"""
+    return any(partial_path in str(path)
+               for partial_path in partial_paths)
+
+
+def pytest_ignore_collect(path, config):
```
Comment (langner) [inline test/conftest.py]: I don't like this for one reason: we're moving the controls for exceptions away from the tests. How are we supposed to keep track of this, presuming we'd eventually like to get rid of these exceptions?

**Comment by berquist** [comment]
This needs more work outside of the PR, so closing until it can be cleaned up with the problems fixed.

**Comment by langner** [comment]
> This needs more work outside of the PR, so closing until it can be cleaned up with the problems fixed.

Could you summarize the remaining work needed in the original issue, if not already there? Or create a project...

**Comment by berquist** [comment]
From highest to lowest importance:

0. There were a lot of unrelated changes that I'll just make small PRs for.
1. Output from `test_all` and the visual tests isn't displayed.
2. Something is wrong with the paths getting sent to the coverage checker, so it looks like the unit and regression tests are covering two different sets of source code.
3. Changing `main` to `test_regressions` allows pytest to automatically run regressions, but the current setup treats it as one big test. It would be nice if each file was its own individual test.
